lib/ansible/module_utils/urls.py

```diff
@@ -1722,6 +1722,66 @@ def url_argument_spec():
     )
 
 
+def normalize_headers(response):
+    if PY2:
+        headers = httplib.HTTPMessage(cStringIO())
+    else:
+        headers = httplib.HTTPMessage()
+
+    if response is None:
+        return headers
+
+    for name, value in response.headers.items():
+        if name in headers:
+            headers[name] = ', '.join((headers[name], value))
+        else:
+            headers[name] = value
+
+    return headers
+
+
+def process_cookies(cookies):
+    # parse the cookies into a nice dictionary
+    cookie_list = []
+    cookie_dict = {}
+    # Python sorts cookies in order of most specific (ie. longest) path first. See ``CookieJar._cookie_attrs``
+    # Cookies with the same path are reversed from response order.
+    # This code makes no assumptions about that, and accepts the order given by python
+    for cookie in cookies:
+        cookie_dict[cookie.name] = cookie.value
+        cookie_list.append((cookie.name, cookie.value))
+    cookies_string = '; '.join('%s=%s' % c for c in cookie_list)
+
+    return cookie_dict, cookies_string
+
+
+def catch_request_errors(url, *args, **kwargs):
```

Comment:
I think ideally this would be better as a context manager.

Reply:
I don't think it's possible to really get the same functionality with a context manager. Otherwise this would need to be rethought.